server/methods/core/payments.js

```diff
@@ -32,16 +32,20 @@ export const methods = {
     // The first record holds the selected billing address
     const billing = cart.billing[0];
     const billingId = Random.id();
+    const shopId = Reaction.getShopId();
     const result = Collection.update({
       _id: id
     }, {
       $addToSet: {
         billing: {
           ...billing,
+          shopId,
```

Comment:
This is not correct. It is causing this to create two billing records even when not in a marketplace scenario. So this second record is incomplete which is why there is no subtotal.